homeassistant/components/nice_go/coordinator.py

```diff
@@ -0,0 +1,215 @@
+
+
+class NiceGOUpdateCoordinator(DataUpdateCoordinator[dict[str, NiceGODevice]]):
+    """DataUpdateCoordinator for Nice G.O."""
+
+    _devices: list[dict[str, Any]] | None = None
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize DataUpdateCoordinator for Nice G.O."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Nice G.O.",
+            update_interval=timedelta(seconds=60),
+        )
+
+        self.refresh_token = self.config_entry.data[CONF_REFRESH_TOKEN]
+        self.refresh_token_creation_time = self.config_entry.data[
+            CONF_REFRESH_TOKEN_CREATION_TIME
+        ]
+        self.email = self.config_entry.data[CONF_EMAIL]
+        self.password = self.config_entry.data[CONF_PASSWORD]
+        self.api = NiceGOApi()
+        self.ws_connected = False
+        self.organization_id: str = ""
+
+    async def _parse_barrier(self, barrier_state: BarrierState) -> NiceGODevice | None:
+        """Parse barrier data."""
+
+        device_id = barrier_state.deviceId
+        name = barrier_state.reported["displayName"]
+        if barrier_state.reported["migrationStatus"] == "NOT_STARTED":
+            ir.async_create_issue(
+                self.hass,
+                DOMAIN,
+                f"firmware_update_required_{device_id}",
+                is_fixable=False,
+                severity=ir.IssueSeverity.ERROR,
+                translation_key="firmware_update_required",
+                translation_placeholders={"device_name": name},
+            )
+            return None
+        ir.async_delete_issue(
+            self.hass, DOMAIN, f"firmware_update_required_{device_id}"
+        )
+        barrier_status_raw = [
+            int(x) for x in barrier_state.reported["barrierStatus"].split(",")
+        ]
+
+        if BARRIER_STATUS[int(barrier_status_raw[2])] == "STATIONARY":
+            barrier_status = "open" if barrier_status_raw[0] == 1 else "closed"
+        else:
+            barrier_status = BARRIER_STATUS[int(barrier_status_raw[2])].lower()
+
+        light_status = barrier_state.reported["lightStatus"].split(",")[0] == "1"
+        fw_version = barrier_state.reported["deviceFwVersion"]
+        if barrier_state.connectionState:
+            connected = barrier_state.connectionState.connected
+        else:
+            connected = False
+
+        return NiceGODevice(
+            id=device_id,
+            name=name,
+            barrier_status=barrier_status,
+            light_status=light_status,
+            fw_version=fw_version,
+            connected=connected,
+        )
+
+    async def _async_update_data(self) -> dict[str, NiceGODevice]:
+        """Fetch data from Nice G.O. API."""
+        async with asyncio.timeout(10):
+            if self.api.id_token:
+                # Already authenticated, returning the same data to avoid overwriting data from the websocket
+                return self.data
```

Comment:
So if I understand correctly the whole setup function is only for setting up for the first time? Use `_async_setup` that's a new feature we added this release